spec/models/feature_flags_spec.rb

```diff
@@ -13,6 +13,30 @@
     end
   end
 
+  describe "global and session flags" do
+    let(:session_data) { { "feature_flags" => {} } }
+
+    context "when global flag is switched on do" do
+      around do |each|
+        ENV["SENTRY_FEATURE_FLAG"] = "enabled"
+        each.run
+        ENV["SENTRY_FEATURE_FLAG"] = "disabled"
+      end
+
+      it "defaults to global flag when the flag does not exist in the session_data" do
+        expect(described_class.enabled?(:sentry, session_data)).to eq true
+      end
+
+      context "when flag is specified in the session_data" do
```

Comment:
would this test be more appropriate if it used a session based flag? Sentry is a global flag that will always look to the env_var and not the session data.
However if we are using sentry flag because we dont want to have to update this test when the more ephemeral flags are removed, im fine with that.

Reply:
> that will always look to the env_var and not the session data

Don't we always look in the session data first? (and for global flags the difference is it normally wouldn't have been put there?)
I think this test does a good job of simulating a session flag at lookup time. I do think we could maybe do with a test of what flags get set in the session data on check creation, though

Reply:
@willc-work yes, the reason I went for the sentry flag (even though it technically wouldn't appear in the session data) is due to it being unlikely that we would need to update the test - that flag should be constant and other flags are temporary and get deleted over time. But I'm definitely open to suggestions of a better way to handle that!

@exonian good shout, I'll see if I can create a test to check which flags get set in the check creation